bundles/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/dhcp/DhcpServerManager.java

```
@@ -49,10 +44,12 @@ public class DhcpServerManager {
     }
 
     public DhcpServerManager(CommandExecutorService service) {
-        if (dhcpServerTool == DhcpServerTool.DNSMASQ) {
+        switch (dhcpServerTool) {
+        case DNSMASQ, NONE:
```

If the tool is `NONE`, then `dnsmasq` is not found in the system. So, maybe it is better to throw an exception.

---

I wanted do that but there are some tests expecting DhcpServerManager is created anyway, so I didn't break some expected behaviours. Btw if we are sure I'll launch an IllegalArgumentException and change the related code.

---

I think that the behavior of the old code is a bit weak, since it assumes that there is a dhcp server installed on the system. We can support the NONE option, if any service if found, make the `linuxTool` optional and manage the fact that it can be empty in the `DhcpServerMonitor`. What do you think?